### PR TITLE
1299 twitter unfollow

### PR DESCRIPTION
**Pull request by Jordon-L**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by Jordon-L** [comment]
#1299

File: components/twitter/actions/unfollow-user/unfollow-user.mjs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    const params = {
+      $,
+      userId,
+      screenName,
+    };
+    const res = await this.twitter.unfollowUser(params);
+    $.export("$summary", "Successfully unfollowed user");
+    return res;
```
Comment (lcaresia) [inline components/twitter/actions/unfollow-user/unfollow-user.mjs:43]: Maybe you can change `res` to `response`, just for verbosity.

```suggestion
    const response = await this.twitter.unfollowUser(params);
    $.export("$summary", "Successfully unfollowed user");
    return response;
```

File: components/twitter/twitter.app.mjs [hunk 2]
```diff
@@ -1076,6 +1076,26 @@ export default {
         config,
       }));
     },
+    async unfollowUser(opts = {}) {
```
Comment (lcaresia) [inline components/twitter/twitter.app.mjs:1079]: You can get the method parameters unstructured here

```suggestion
    async unfollowUser({params, $}) {
```

File: components/twitter/twitter.app.mjs [hunk 3]
```diff
@@ -1076,6 +1076,26 @@ export default {
...
+      const config = {
+        url: "https://api.twitter.com/1.1/friendships/destroy.json",
+        method: "post",
+        params,
+      };
+      return (await this._makeRequest({
+        $,
+        config,
+      }));
+    },
```
Comment (lcaresia) [inline components/twitter/twitter.app.mjs:1098]:
```suggestion
      const config = {
        url: "https://api.twitter.com/1.1/friendships/destroy.json",
        method: "post",
        params,
      };
      return (await this._makeRequest({
        $,
        config,
      }));
    },
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Looks very good, I requested just a few things.

File: components/twitter/actions/unfollow-user/unfollow-user.mjs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import twitter from "../../twitter.app.mjs";
```
Comment (lcaresia) [inline components/twitter/actions/unfollow-user/unfollow-user.mjs:1]:
```suggestion
import twitter from "../../twitter.app.mjs";
import { ConfigurationError } from "@pipedream/platform";
```

**Review by lcaresia — CHANGES_REQUESTED** [review]

**Comment by Jordon-L** [comment]
I have encountered an error after adding in the suggestions. I'll see if I can fix it.

**Comment by Jordon-L** [comment]
I have fixed the error. I was missing one of the suggestions.

**Comment by dylburger** [comment]
Thanks @Jordon-L , I moved this back into the **Ready for PR Review** column of our backlog again, and someone should validate the changes you made. Then we'll do a quick QA !

**Comment by Jordon-L** [comment]
Ok. Let me know If I need to change something.

File: components/twitter/twitter.app.mjs (hunk 2 above)
Comment (andrewjschuang) [reply components/twitter/twitter.app.mjs]: @Jordon-L @lcaresia seems to have two extra spaces before `params` and `$`

File: components/twitter/twitter.app.mjs [hunk 5]
```diff
@@ -1076,6 +1076,20 @@ export default {
         config,
       }));
     },
+    async unfollowUser({
+        params,
+        $,
+    }) {
```
Comment (andrewjschuang) [inline components/twitter/twitter.app.mjs:1082]: **ESLint**

```suggestion
    async unfollowUser({
      params,
      $,
    }) {
```

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
@Jordon-L final change! Just two extra spaces that were in the suggestion.

I've move this card to the To Do Column, when you're done you can move this back to Ready for PR Review!

**Comment by Jordon-L** [comment]
Thanks.

**Comment by Jordon-L** [comment]
How do I move this card to Ready for PR Review?

**Comment by andrewjschuang** [comment]
> How do I move this card to Ready for PR Review?

One way is through here:
Screen Shot 2022-05-23 at 9 39 47 AM

**Review by andrewjschuang — APPROVED** [review]
@Jordon-L thanks for the contribution! LGTM to QA!

**Comment by Hassan-Abbas-SQA** [comment]
if userId and screenName is required why they are optional fields instead of required fields? 

image

**Comment by andrewjschuang** [comment]
> if userId and screenName is required why they are optional fields instead of required fields?

I think it requires either one of them, and not both

**Comment by Hassan-Abbas-SQA** [comment]
yes it requires either one of them but it should be written that either one of them is required

**Comment by andrewjschuang** [comment]
@Jordon-L thanks for the contribution! It's merged now!

**Pull request merged**